cat-shark-carousel-server/src/index.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'path';
+import express, { Application, NextFunction, Request, Response } from 'express';
+import { catsList, sharksList } from './database';
+import { shuffle } from './util';
+
+const app: Application = express();
+const port = process.env.PORT || 3000;
+
+enum PhotoTypes {
+    SHARK = "SHARK",
+    CAT = "CAT",
+};
+
+const mapping = {
+    [PhotoTypes.SHARK]: sharksList,
+    [PhotoTypes.CAT]: catsList,
+}
+
+app.use((req: Request, res: Response, next: NextFunction) => {
+    res.setHeader("Access-Control-Allow-Origin", "*");
+    res.setHeader("Access-Control-Allow-Methods", "GET");
+    res.setHeader("Access-Control-Allow-Headers", "*");
+    next();
+})
+
+app.get('/photos', (req: Request, res: Response) => {
+    let photoTypes: undefined | string | string[] = req.query.types as string[];
+    if (!photoTypes) {
+        photoTypes = [];
+    } else if (!Array.isArray(photoTypes)) {
+        photoTypes = [photoTypes];
+    }
+    const photoList = photoTypes.filter((value) => Object.values(PhotoTypes).includes(value as PhotoTypes))
+        .map((value) => mapping[value as PhotoTypes])
+        .flat();
+
+    shuffle(photoList);
+
+    res.setHeader('Content-Type', 'application/json');
+    res.status(200).json(photoList).send();
+})
```

Comment:
Can you explain your API design choice? Are there any changes you could make to make it "more RESTful"?

Reply:
Sure, when thinking how to create the API, I came to the conclusion that it would be wasteful to create anymore then 1 API endpoint as the data that I would be returning would be of the same type (photo URLs) but they would be able to be sorted as different categories (animal type). To allow for this filtering, I’d allow a user to pass in the types of animals they’d like to see. This would put it on the user to decide the type of list they’d want.

However, now that I’m thinking about it, a “more RESTful” way of doing this would be to return a full list (a default GET should return “all” of the information) and then I’d let the user “filter” out the type of information they don’t want (ex. sharks and/or cats).